astropy/visualization/interval.py

```diff
@@ -101,19 +101,22 @@ def __init__(self, lower_percentile, upper_percentile, n_samples=None):
     def get_limits(self, values):
 
         # Make sure values is a Numpy array
-        values = np.asarray(values)
+        values = np.asarray(values).ravel()
 
         # If needed, limit the number of samples. We sample with replacement
         # since this is much faster.
         if self.n_samples is not None and values.size > self.n_samples:
-            values = np.random.choice(values, self.n_samples)
+            try:
+                values = np.random.choice(values, self.n_samples)
+            except AttributeError:  # Numpy 1.6.x
+                values = values[np.random.randint(0, values.size, self.n_samples)]
```

Comment:
What's interesting is that with this workaround the test still works which means that ``np.random.choice`` is doing exactly this behind the scenes!